src/os/mapcontainer.js

```diff
@@ -285,11 +292,72 @@ os.MapContainer.prototype.getWebGLRenderer = function() {
 
 
 /**
- * Set the WebGL renderer.
+ * Get the available WebGL renderers.
+ *
+ * @return {Object<string, os.webgl.IWebGLRenderer>}
+ */
+os.MapContainer.prototype.getWebGLRenderers = function() {
+  return this.webGLRenderers_;
+};
+
+
+/**
+ * Get the key for a renderer
+ * Effectively the `id` for instances of {@link os.webgl.AbstractWebGLRenderer}
+ *
+ * @param {os.webgl.IWebGLRenderer} renderer
+ * @return {string|undefined}
+ * @private
+ */
+os.MapContainer.prototype.getWebGLRendererKey_ = function(renderer) {
+  if (this.webGLRenderers_) {
+    for (const [key, val] of Object.entries(this.webGLRenderers_)) {
+      if (renderer == val) {
+        return /** @type {string|undefined} */ (key);
+      }
+    }
+  }
+  return undefined;
+};
+
+
+/**
+ * Add a WebGL renderer
+ *
+ * @param {os.webgl.IWebGLRenderer|undefined} renderer
+ */
+os.MapContainer.prototype.addWebGLRenderer = function(renderer) {
+  if (renderer) {
+    this.webGLRenderers_[renderer.id || os.string.randomString()] = renderer;
+  }
+};
+
+
+/**
+ * Remove a WebGL renderer
+ *
+ * @param {os.webgl.IWebGLRenderer} renderer
+ * @return {boolean}
+ */
+os.MapContainer.prototype.removeWebGLRenderer = function(renderer) {
```

Comment:
Is there a use case for this? Seems like this and `getWebGLRendererKey_` wouldn't typically be used (plugins will either add the renderer or not on init).